### PR TITLE
Fix(Core/WorldBoss): Azuregos rewrite

### PR DESCRIPTION
**Pull request by Nefertumm**
## Changes Proposed:
-  Rewrite the whole fight.
-  Added waypoints and gossip.
-  Added waypoints for Spirit of Azuregos too.
-  Fixed mechanic for Mark of Frost
-  Use the correct Cleave and Arcane Vacuum spell.
-  Use the actual spell for teleporting the players under Azuregos.
-  Waypoints are cherrypicked from vMangos. I couldn't find the author tho :(

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/10336
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/10337
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/10175
- Closes https://github.com/chromiecraft/chromiecraft/issues/2788

## Tests Performed:

- Tested in-game

## How to Test the Changes:

1. Go Azuregos
2. See his path and check his fight.

## Known Issues and TODO List:

- [x] Re-do the methods for remove auras based on Winfi's suggestions.
- [ ] ??? 

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Maps/Map.cpp [hunk 1]
```diff
@@ -3757,6 +3757,51 @@ void Map::SetZoneOverrideLight(uint32 zoneId, uint32 lightId, Milliseconds fadeI
...
+                if (!player->IsGameMaster() && ((zoneId && player->GetZoneId() == zoneId) || !zoneId))
+                {
+                    player->RemoveAurasDueToSpell(spellId);
+                }
+            }
+        }
+    }
+}
+
+void Map::RemoveAurasDueToSpellsOnPlayersInZone(std::list<uint32> spellIds, uint32 zoneId /*= 0*/)
```
Comment (Kitzunu) [inline src/server/game/Maps/Map.cpp:3779]: can't you just make these two functions to one and allow a list?

File: src/server/game/Spells/SpellMgr.cpp [hunk 2]
```diff
@@ -3575,6 +3575,12 @@ void SpellMgr::LoadDbcDataCorrections()
         spellInfo->RangeIndex = 6; // 100 yards
     });
 
+    ApplySpellFix({ 21147 }, [](SpellEntry* spellInfo)
```
Comment (Kitzunu) [inline src/server/game/Spells/SpellMgr.cpp:3578]: add comment

File: src/server/scripts/Kalimdor/boss_azuregos.cpp [hunk 3]
```diff
@@ -15,156 +15,186 @@
...
 public:
-    boss_azuregos() : CreatureScript("boss_azuregos") { }
 
-    CreatureAI* GetAI(Creature* creature) const override
-    {
-        return new boss_azuregosAI(creature);
-    }
+    boss_azuregos() : CreatureScript("boss_azuregos") { }
 
     struct boss_azuregosAI : public ScriptedAI
```
Comment (Kitzunu) [inline src/server/scripts/Kalimdor/boss_azuregos.cpp:49]: might as well convert to new system while you rework the entire script :P

**Comment by Nefertumm** [comment]
Currently waiting for https://github.com/azerothcore/azerothcore-wotlk/pull/10371 to be merged. Do not merge before.

**Comment by temperrr** [comment]
Tested, 

Works as described

However, not sure if Mark of Frost should be removed after wiping. I can't find proof for it either way

**Review by Nyeriah — APPROVED** [review]

**Review by Kitzunu — CHANGES_REQUESTED** [review]

**Comment by Nyeriah** [comment]
@Nefertumm check kitzunu's requests please, gotta merge this saturday

**Comment by Nefertumm** [comment]
Done

File: src/server/scripts/Kalimdor/boss_azuregos.cpp [hunk 4]
```diff
@@ -15,156 +15,220 @@
...
+
+// Mark of Frost - Triggered Spell
+class spell_mark_of_frost_freeze : public SpellScript
+{
+    PrepareSpellScript(spell_mark_of_frost_freeze);
 
-            DoMeleeAttackIfReady();
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_MARK_OF_FROST, SPELL_AURA_OF_FROST, SPELL_MARK_OF_FROST_AURA });
```
Comment (Kitzunu) [inline src/server/scripts/Kalimdor/boss_azuregos.cpp:210]:
```suggestion
        return ValidateSpellInfo({ SPELL_MARK_OF_FROST, SPELL_AURA_OF_FROST });
```
Only validate spells you use inside the script

File: src/server/scripts/Kalimdor/boss_azuregos.cpp [hunk 5]
```diff
@@ -15,156 +15,220 @@
...
-            //CleaveTimer
-            if (CleaveTimer <= diff)
-            {
-                DoCastVictim(SPELL_CLEAVE);
-                CleaveTimer = 7000;
-            }
-            else CleaveTimer -= diff;
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_ARCANE_VACUUM, SPELL_ARCANE_VACUUM_TP });
```
Comment (Kitzunu) [inline src/server/scripts/Kalimdor/boss_azuregos.cpp:184]: here too

**Review by Kitzunu — APPROVED** [review]

**Pull request merged**